From b8fc94cdb144467d88f35344076fd3621af93a17 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Wed, 7 Jul 2021 14:56:44 +0900
Subject: [PATCH] ksmbd: fix read on the uninitialized send_ctx
Git-commit: b8fc94cdb144467d88f35344076fd3621af93a17
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


If st->status is not SMB_DIRECT_CS_CONNECTED, It will jump done label
and accessing the uninitialized send_ctxi by smb_direct_flush_send_list
will cause kernel oops. This patch just return -ENOTCONN to avoid it.

Reported-by: Coverity Scan <scan-admin@coverity.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/ksmbd/transport_rdma.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/fs/ksmbd/transport_rdma.c b/fs/ksmbd/transport_rdma.c
index 171fb3dd018a..d5728c84a15a 100644
--- a/fs/ksmbd/transport_rdma.c
+++ b/fs/ksmbd/transport_rdma.c
@@ -1207,10 +1207,8 @@ static int smb_direct_writev(struct ksmbd_transport *t,
 	struct kvec vec;
 	struct smb_direct_send_ctx send_ctx;
 
-	if (st->status != SMB_DIRECT_CS_CONNECTED) {
-		ret = -ENOTCONN;
-		goto done;
-	}
+	if (st->status != SMB_DIRECT_CS_CONNECTED)
+		return -ENOTCONN;
 
 	//FIXME: skip RFC1002 header..
 	buflen -= 4;
-- 
2.33.0

